kernel/aster-nix/src/arch/x86/cpu.rs

```diff
@@ -100,3 +101,26 @@ impl GpRegs {
         copy_gp_regs!(src, self);
     }
 }
+
+impl PageFaultContext for CpuExceptionInfo {
+    fn is_page_fault(&self) -> bool {
+        self.cpu_exception() == PAGE_FAULT
+    }
+
+    fn page_fault_address(&self) -> Vaddr {
+        self.page_fault_addr
+    }
+
+    fn page_fault_required_perms(&self) -> VmPerms {
+        const WRITE_ACCESS_MASK: usize = 0x1 << 1;
+        const INSTRUCTION_FETCH_MASK: usize = 0x1 << 4;
+
+        if self.error_code & INSTRUCTION_FETCH_MASK != 0 {
+            VmPerms::EXEC
+        } else if self.error_code & WRITE_ACCESS_MASK != 0 {
+            VmPerms::WRITE
+        } else {
+            VmPerms::READ
+        }
+    }
+}
```

Comment:
The interface of this trait is a bit self-contradictory. If a `CpuExeptionInfo` can have `self.is_page_fault == false`, it too provides a `.page_fault_address`.

Maybe we can introduce a structure `PageFaultInfo { addr, missing_perms }` and `impl TryInto<PageFaultInfo> for CpuExceptionInfo`?